src/main/java/com/minecolonies/core/tileentities/TileEntityColonyBuilding.java

```diff
@@ -595,6 +597,18 @@ public IItemHandler getItemHandlerCap(final Direction side)
                                 {
```

Comment:
Remove the whole block from line 582 to 623 and replace it with the utility method mentioned in the other comment.